Automattic/studio
# Avoid running wp-cli to get themes and plugins when adding or importing a site

- Closes https://github.com/Automattic/dotcom-forge/issues/8625#issuecomment-2285612436

## Proposed Changes

- Extend the isAddingSite state until the site starts and it has the proper file structure
- Avoid running wp-cli commands to get themes and plugins information while a site is being added or imported
- Refactor createSite from useSiteDetails

Currently there are two wp-cli errors being output in the console:
The first one is when getting the themes and plugins context for the Assistant, and this error is fixed in this PR.
The second one is produced when running the search replace command, and it needs a second PR to get fixed. See https://github.com/Automattic/studio/pull/437#issuecomment-2286057665

## Testing Instructions



~Apply this diff to avoid displaying the second error that this PR does not fix:~
```
diff --git a/src/lib/import-export/import/importers/importer.ts b/src/lib/import-export/import/importers/importer.ts
index a541923..0305eb2 100644
--- a/src/lib/import-export/import/importers/importer.ts
+++ b/src/lib/import-export/import/importers/importer.ts
@@ -65,7 +65,7 @@ abstract class BaseImporter extends EventEmitter implements Importer {
                        }
                }
 
-               await this.replaceSiteUrl( siteId );
+               // await this.replaceSiteUrl( siteId );
                this.emit( ImportEvents.IMPORT_DATABASE_COMPLETE );
        }
 
```

- Run `STUDIO_IMPORT_EXPORT=true npm start`
- Click on Add site in the left sidebar
- Select a backup
- Click add site
- Observe the console in NodeJS and confirm there are no errors
- Repeat the steps now without importing any site to confirm the flow is still working




https://github.com/user-attachments/assets/e0e0f3b3-e6f7-4620-9e8a-c7aa836c2c4e



## Pre-merge Checklist
